### DISCUSSION TITLE
Add addon dynamically

### DISCUSSION THREAD

**Discussion by anylayer** [mitmproxy/mitmproxy#5338] [General]
How to add addon plugin dynamically  when DumpMaster is running.

**Comment by mhils** [comment]
Two possible approaches:

1. Add it explicitly from your Python code.
```python
ctx.master.addons.add(MyAddon())
```

2. If you run `mitmdump -s script.py`, you can update the contents of `addons` in `script.py`, which will then be picked up by live-reload (https://docs.mitmproxy.org/dev/addons-overview/):

`script.py`
```python
...

addons = [Counter()]  # <- add more addons here and wait for live reload.
```

**Comment by ofir-ps** [comment]
Is it possible to add addons dynamically per flow? For example, add an addon to a flow based on request/response url or header
Alternatively, is it possible for an addon to skip the next addons for a specific flow? Do some sort of short-circuit and bypass the rest?
